contracts/MandatoryLockupDistributor.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.4;
+
+import "OpenZeppelin/openzeppelin-contracts@4.6.0/contracts/utils/cryptography/MerkleProof.sol";
+import "./AbstractLockupDistributor.sol";
+
+interface IOGVStaking {
+    function stake(
+        uint256 amount,
+        uint256 end,
+        address to
+    ) external;
+}
+
+contract MandatoryLockupDistributor is AbstractLockupDistributor {
+
+    constructor(
+        address _token,
+        bytes32 _merkleRoot,
+        address _stakingContract,
+        uint256 _endBlock
+    ) AbstractLockupDistributor(_token, _merkleRoot, _stakingContract, _endBlock) {}
+
+    /**
+     * @dev Execute a claim using a merkle proof with optional lockup in the staking contract.
+     * @param _index Index in the tree
+     * @param _amount Amount eligiblle to claim
+     * @param _merkleProof The proof
+     */
+    function claim(
+        uint256 _index,
+        uint256 _amount,
+        bytes32[] calldata _merkleProof
+    ) external {
+        require(!isClaimed(_index), "MerkleDistributor: Drop already claimed.");
```

Comment:
Just checking a design decision here. These claim methods don't stop working after the endBlock, it's currently more a soft end until we burn the funds and the staking calls break on balance transfer errors.


Do we want to just have a `require(block.number < endBlock)` in these claim methods, so the end requires no human action, and is explicit and precise?

Reply:
Sounds like a good proposal to me. Adding that in!